### ISSUE TITLE
fetchbower: ability to fetch local package

### ISSUE THREAD

**Issue by domenkozar** [NixOS/nixpkgs#19644]
[bower2nix](https://github.com/rvl/bower2nix/issues/7) now supports parsing relative paths in `bower.json` dependency specification.

I wonder how we should support that at `fetchbower`? Or maybe it's more feasible to chain `buildBowerComponents`, but somehow `bower2nix` should support it.

The common use case is to have two bower packages in the same source repository.

cc @shlevy @rvl

**Comment by rvl** [comment]
I agree it's easier to use git submodules than support authentication.

Looks like another function might do the job:
```
(fetchbowerLocal "foo" ../foo)
```
This would cause a bower install of `/nix/store/...-foo`. The deps of foo have already been generated and appear in the same nix file.

Does that sound like it could work?

Whatever we do, I don't think it should be too clever, because many new projects use webpack or systemjs rather than bower.

**Comment by domenkozar** [comment]
@rvl so one would need to run `bower2nix` on each relative path before that?

**Comment by rvl** [comment]
No, hopefully not. I have changed `bower2nix` so that it can recursively generate dependencies of the relative paths. All that remains is to be able to fetch `../foo`.

**Comment by ixmatus** [comment]
Is there an update on this? I'm currently running into this issue.

**Comment by ixmatus** [comment]
Is there work-in-progress I could help with or take advantage of? @rvl
